go/libraries/doltcore/schema/index.go

```diff
@@ -68,12 +68,17 @@ type indexImpl struct {
 	comment       string
 }
 
-func NewIndex(name string, tags, allTags []uint64, indexColl *indexCollectionImpl, props IndexProperties) Index {
+func NewIndex(name string, tags, allTags []uint64, indexColl IndexCollection, props IndexProperties) Index {
+	var idxColl *indexCollectionImpl
```

Comment:
Need a better name. `indexColl` and `idxColl` are the same name. Something like `indexCollImpl`.